drkafka/src/test/java/com/pinterest/doctorkafka/KafkaClusterTest.java

```diff
@@ -0,0 +1,117 @@
+    ConcurrentMap<String, ConcurrentMap<TopicPartition, Histogram>> oldBytesOutStats = ReplicaStatsManager.bytesOutStats;
+
+    TopicPartition topicPartition = new TopicPartition(TOPIC, 0);
+
+    ConcurrentMap<String, ConcurrentMap<TopicPartition, Histogram>> testBytesInStats = new ConcurrentHashMap<>();
+    ConcurrentMap<String, ConcurrentMap<TopicPartition, Histogram>> testBytesOutStats = new ConcurrentHashMap<>();
+    ConcurrentMap<TopicPartition, Histogram> testBytesInHistograms = new ConcurrentHashMap<>();
+    ConcurrentMap<TopicPartition, Histogram> testBytesOutHistograms = new ConcurrentHashMap<>();
+    Histogram inHist = new Histogram(new UniformReservoir());
+    Histogram outHist = new Histogram(new UniformReservoir());
+
+    inHist.update(0);
+    outHist.update(0);
+
+    testBytesInHistograms.put(topicPartition, inHist);
+    testBytesOutHistograms.put(topicPartition, outHist);
+
+    testBytesInStats.put(ZOOKEEPER_URL, testBytesInHistograms);
+    testBytesOutStats.put(ZOOKEEPER_URL, testBytesOutHistograms);
+
+    ReplicaStatsManager.bytesInStats = testBytesInStats;
+    ReplicaStatsManager.bytesOutStats = testBytesOutStats;
+
+    KafkaCluster kafkaCluster = new KafkaCluster(ZOOKEEPER_URL, doctorKafkaClusterConfig);
+
+    Node[] nodes = new Node[]{
+        new Node(0, "test00", 9092),
+        new Node(1, "test01", 9092),
+        new Node(2, "test02", 9092),
+        new Node(3, "test03", 9092),
+        new Node(4, "test04", 9092)
+    };
+
+    Node leader = nodes[0];
+    Node[] replicas = new Node[]{
+        nodes[0],
+        nodes[1],
+        nodes[2]
+    };
+
+    Node[] isrs = new Node[]{
+        nodes[0]
+    };
+    PartitionInfo partitionInfo = new PartitionInfo(TOPIC, 0, leader, replicas, isrs);
+    OutOfSyncReplica oosReplica = new OutOfSyncReplica(partitionInfo);
+    oosReplica.replicaBrokers = Arrays.asList(new Integer[]{0,1,2});
+
+
+    KafkaBroker[] brokers = new KafkaBroker[]{
+        new KafkaBroker(doctorKafkaClusterConfig, 0),
+        new KafkaBroker(doctorKafkaClusterConfig, 1),
+        new KafkaBroker(doctorKafkaClusterConfig, 2),
+        new KafkaBroker(doctorKafkaClusterConfig, 3),
+        new KafkaBroker(doctorKafkaClusterConfig, 4)
+    };
+
+    // setting the reservedBytesIn for priority queue comparisons
+    brokers[4].reserveInBoundBandwidth(topicPartition, 10);
+    brokers[4].reserveOutBoundBandwidth(topicPartition, 10);
+
+    PriorityQueue<KafkaBroker> brokerQueue = new PriorityQueue<>();
+
+    for ( KafkaBroker broker : brokers){
+      brokerQueue.add(broker);
+    }
+
+    int beforeSize = brokerQueue.size();
+
+    /**
+     * Since the BytesInStats/BytesOutStats are not set,
+     * getMaxBytesIn() and getMaxBytesOut() will return 0 for each broker,
+     * so brokerQueue won't change
+     **/
+    Map<Integer, KafkaBroker> altBrokers = kafkaCluster.getAlternativeBrokers(brokerQueue, oosReplica);
+
+    assertNotNull(altBrokers);
```

Comment:
add some comments to explain why these invariants hold?  that will be helpful to other readers.